packages/api/src/useSubscription.ts

```diff
@@ -1,42 +1,62 @@
-import { useState } from 'react';
-import { useWS as useWSShared } from '@deriv/shared';
-import { TSocketSubscribableEndpointNames, TSocketAcceptableProps, TSocketResponseData } from '../types';
+import { useCallback, useEffect, useRef, useState } from 'react';
+import { useWS } from '@deriv/shared';
+import type {
+    TSocketSubscribableEndpointNames,
+    TSocketAcceptableProps,
+    TSocketResponseData,
+    TSocketRequestCleaned,
+} from '../types';
 
 const useSubscription = <T extends TSocketSubscribableEndpointNames>(name: T) => {
     const [is_loading, setIsLoading] = useState(false);
     const [is_subscribed, setSubscribed] = useState(false);
     const [error, setError] = useState<unknown>();
```

Comment:
```suggestion
    const [error, setError] = useState<unknown>();
```
I think we have some basic structure for errors 🤔 IIRC 
```
type Error = {
   error: string;
   code: string;
}
```

Reply:
@yashim-deriv This can also be an exception, That's why we have to keep the type as `unknown` to force devs to do a type check before using the error object in their code.